tools/c7n_azure/c7n_azure/resources/cdn_custom_domain.py

```diff
@@ -58,3 +58,49 @@ def extra_args(cls, parent_resource):
                 'endpoint_name': endpoint_name,
                 'profile_name': profile_name
             }
+
+
+@CdnCustomDomain.filter_registry.register('tls-version')
+class CdnManagedHttpsParametersFilter(ValueFilter):
+    """Find all Cdn Endpoint Custom Domain with minimum Tls Version set to TLS12
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: cdn-endpoint-custom-domain-tls-version
+            resource: azure.cdn-custom-domain
+            filters:
+               - and:
+                  - type: value
+                    key: properties.customHttpsProvisioningState
+                    op: eq
+                    value: Enabled
+                  - type: tls-version
+                    key: properties.customHttpsParameters.minimumTlsVersion
+                    op: eq
+                    value: "TLS12"
+
+    """
+    schema = type_schema(
+        'tls-version',
+        rinherit=ValueFilter.schema,
+        value = {'type': 'string', 'enum':['None', 'TLS10', 'TLS12']})
+
+    def process(self, resources, event=None):
+        desired_version = self.data['value']
+        for custom_domain in resources:
+            properties = custom_domain.get('properties', {})
+            if 'customHttpsParameters' not in properties:
+                http_params = {
+                      "certificateSource": "Cdn",
+                      "protocolType": "ServerNameIndication",
+                      "minimumTlsVersion": desired_version,
```

Comment:
shouldn't the default here be what the service does not what the policy author specifies?

Reply:
Updated to reflect the service offering instead of policy author's specification.